pkg/apis/eventing/v1alpha1/channel_types.go

```diff
@@ -117,6 +117,16 @@ func (cs *ChannelStatus) GetCondition(t duckv1alpha1.ConditionType) *duckv1alpha
 	return chanCondSet.Manage(cs).GetCondition(t)
 }
 
+// IsReady returns true if the resource is ready overall.
+func (cs *ChannelStatus) IsReady() bool {
+	return chanCondSet.Manage(cs).IsHappy()
+}
+
+// InitializeConditions sets relevant unset conditions to Unknown state.
+func (cs *ChannelStatus) InitializeConditions() {
```

Comment:
Just FYI, I added `InitializeConditions` in my PR #468 along with a test.

Reply:
Oh nice!! can you also add isReady to #468 and I will close this...